src/components/routes/HomePage.tsx

```diff
@@ -18,33 +18,50 @@ export default function HomePage() {
 
   useEffect(() => {
     const urlSearchParams = new URLSearchParams(window.location.search);
-    const importJSON = urlSearchParams.get("importJSON");
+    const importJSON = urlSearchParams.has("postMessage");
 
     if (importJSON) {
-      try {
-        const parsed = JSON.parse(importJSON) as Record<string, unknown>;
-        const result = deserialize(parsed);
+      // post back a ready message
+      (window.opener as WindowProxy | null)?.postMessage(
+        "ready",
+        "https://fxms-skylight.netlify.app/"
+      );
```

Comment:
add a `types/global.d.ts` with the following:

```
interface Window {
  opener: WindowProxy | null;
}
```

Reply:
Then you should be able to use `window.opener?.postMessage` directly

Reply:
I am going to file a ticket to get this in, because it's giving me some issues atm. Maybe I can do Zod at the same time :P